src/core/project/qgsprojectgpssettings.cpp

```diff
@@ -85,6 +106,18 @@ QDomElement QgsProjectGpsSettings::writeXml( QDomDocument &doc, const QgsReadWri
     element.setAttribute( QStringLiteral( "destinationLayer" ), QString() );
   }
 
+  {
+    QDomElement timeStampElement = doc.createElement( QStringLiteral( "timeStampFields" ) );
```

Comment:
What about doing a bit of sanitation here and avoid saving details for a layer that's been removed (i.e. a layer ID that wouldn't match any project map layers anymore)?

Reply:
Ok, done!

Reply:
@nyalldawson , thanks.